.github/workflows/benchstat-pr.yml

```diff
@@ -0,0 +1,75 @@
+name: benchstat
+
+on: [pull_request]
+
+jobs:
+  benchstat:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
+
+      # Generate benchmark report for main branch
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: main
+      - name: Benchmark
+        run: go test ./... -count=10 -run=Benchmark -bench=. | tee -a bench.txt
```

Comment:
I imagine you're passing `-run=Benchmark` to skip running any tests, however, if for some reason you have a test function with `Benchmark` anywhere in the name it will run that:

```go
package main

import "testing"

func TestBenchmarkOne(t *testing.T) {
	t.Fatal("shouldn't have run")
}

func BenchmarkOne(b *testing.B) {
	for i :=0;i<b.N;i++ {
		_ = i
	}
}
```

If you run this you get the following:

```
$ go test -run=Benchmark -bench=. .
--- FAIL: TestBenchmarkOne (0.00s)
    misc_test.go:6: shouldn't have run
FAIL
exit status 1
```

Perhaps it's better to change it to `-run'^$'` which won't run any tests at all.

Reply:
Also, let's add `-benchmem` to it so it not only benchmark the CPU performance but also the memory consumption and allocations.